demos/portal-embed-demo/public/js/ttt/audio-controller.js

```diff
@@ -0,0 +1,369 @@
+
+/**
+ * Audio Controller
+ * Controlling all the audio related features
+ */
+class AudioController extends HTMLElement {
+    
+    /**
+     * Initiate the element
+     */
+    connectedCallback() {
+        // Attach to Shadow DOM
+        this.root = this.attachShadow({mode: 'open'});
+        this.root.innerHTML = `<style>${this.css}</style>${this.template}`;
+
+        // All the UI references
+        this.player = this.root;
+        this.playButton = this.root.querySelector('.play');
+        this.nextButton = this.root.querySelector('.next');
+        this.prevButton = this.root.querySelector('.prev');
+        this.timeline = this.root.querySelector('.timeline');
+        this.axis = this.root.querySelector('.axis');
+        this.audioTitle = this.root.querySelector('.title');
+
+        // Initiate the audio player
+        const audioId = this.getAttribute('data-audioid');
+        this._createNewAudio(audioId);
+        this.playlist = [audioId];
+        this.playlistIndex = 0;
+        this.isPlaying = false;
+
+        // Add event listeners
+        this._hookEvents();
+    }
+
+    /**
+     * CSS
+     */
+    get css() {
+        return `
+            :host {
+                width: 100%;
+                height: 80px;
+                background-color: slategray;
+                position: fixed;
+                bottom: 0px;
+                z-index: 999;
+                display: block;
+            }
+            .controllers {
+                margin: 0 auto;
+                height:80%;
+                width:180px;
+            }
+            .controller {
+                width: 60px;
+                height: 80%;
+                text-align: center;
+                float: left;
+            }
+            .controller img{
+                height: 50%;
+                cursor: pointer;
+            }
+            .flip-horizontal {
+                transform: scale(-1, 1);
+            }
+            .title {
+                width: 100%;
+                height: 20%;
+                text-align: center;
+                margin-top: 8px;
+                font-size: 0.8em;
+                color: #FFF;
+            }
+            .timeline {
+                width: 90%;
+                max-width: 550px;
+                border-top: 1px solid #FFF;
+                margin: 8px auto 3px auto;
+                cursor: pointer;
+            }
+            .axis{
+                width: 10px;
+                height: 10px;
+                background-color: #FFF;
+                position: relative;
+                left: 0px;
+                top: -5px;
+                border-radius: 50%;
+                cursor: pointer;
+            }
+            .prev {
+                opacity: 0.3;
+            }
+            .next {
+                opacity: 0.3;
+            }
+        `;
+    }
+
+    /**
+     * HTML template
+     */
+    get template() {
+        return `
+            <div class='title'></div>
```

Comment:
nit: double-quotes are usually used for HTML attributes (and you're inside backticks, so you should be able to use them without extra escaping)

Reply:
Thank you for this! I changed all (including ttt-reco, writer-follow) the template() HTML using double quotes!